src/scripts/web3-interface.ts

```diff
@@ -1,90 +1,58 @@
+import { SarcoSupportedNetwork } from "@sarcophagus-org/sarcophagus-v2-sdk";
 
 export interface Web3Interface {
-  networkName: string;
-  ethWallet: ethers.Wallet;
-  encryptionHdWallet: ethers.utils.HDNode;
-  keyFinder: KeyFinder;
-  sarcoToken: IERC20;
-  archaeologistFacet: ArchaeologistFacetX;
-  embalmerFacet: EmbalmerFacet;
-  thirdPartyFacet: ThirdPartyFacet;
-  viewStateFacet: ViewStateFacet;
-  networkConfig: NetworkConfig;
+  networkContexts: NetworkContext[];
+  getNetworkContext: (network: SarcoSupportedNetwork | undefined) => NetworkContext;
 }
 
 let web3Interface: Web3Interface | undefined;
 
 export const destroyWeb3Interface = async (): Promise<void> => {
   if (!!web3Interface) {
-    web3Interface.ethWallet.provider.removeAllListeners();
-    (web3Interface.ethWallet.provider as ethers.providers.WebSocketProvider)._websocket.terminate();
+    web3Interface.networkContexts.forEach(networkContext => {
+      networkContext.ethWallet.provider.removeAllListeners();
+      (
+        networkContext.ethWallet.provider as ethers.providers.WebSocketProvider
+      )._websocket.terminate();
+    });
     web3Interface = undefined;
   }
 };
 
-export const getWeb3Interface = async (isTest?: boolean): Promise<Web3Interface> => {
+export const getWeb3Interface = async (isTest: boolean = false): Promise<Web3Interface> => {
   if (!!web3Interface) {
     return web3Interface;
   }
 
   try {
-    const networkConfig = getNetworkConfigByChainId(process.env.CHAIN_ID || localChainId);
-    // PROVIDER_URL should be "wss://<network>.infura.io/ws/v3/<api-key>"
-    const rpcProvider = new ethers.providers.WebSocketProvider(process.env.PROVIDER_URL!);
-
-    const ethWallet = isTest
-      ? ethers.Wallet.createRandom()
-      : new ethers.Wallet(process.env.ETH_PRIVATE_KEY!, rpcProvider);
-
-    const encryptionHdWallet = ethers.utils.HDNode.fromMnemonic(process.env.ENCRYPTION_MNEMONIC!);
-
-    const network = await rpcProvider.detectNetwork();
-
-    const sarcoToken = IERC20__factory.connect(networkConfig.sarcoTokenAddress, ethWallet);
+    const chainIds = process.env.CHAIN_IDS
+      ? process.env.CHAIN_IDS.split(",").map(chaidIdStr => Number(chaidIdStr.trim()))
+      : [];
+    const networkContexts = getNetworkContextsByChainIds(chainIds, isTest);
 
-    const diamondDeployAddress =
-      process.env.DIAMOND_DEPLOY_ADDRESS ?? networkConfig.diamondDeployAddress;
-
-    const archaeologistFacet = ArchaeologistFacet__factory.connect(diamondDeployAddress, ethWallet);
-    const viewStateFacet = ViewStateFacet__factory.connect(diamondDeployAddress, ethWallet);
-    const thirdPartyFacet = ThirdPartyFacet__factory.connect(diamondDeployAddress, ethWallet);
-    const embalmerFacet = EmbalmerFacet__factory.connect(diamondDeployAddress, ethWallet);
-
-    const keyFinder = new KeyFinder(encryptionHdWallet);
+    web3Interface = {
+      networkContexts,
+      getNetworkContext: networkOrChainId => {
```

Comment:
Does this need to support both network name and chainId? Could we standardize it to just support one?